### DISCUSSION TITLE
Can I change choice of choice node from a different non-choide node?

### DISCUSSION THREAD

**Discussion by ziontee113** [L3MON4D3/LuaSnip#638] [Q&A]
Hi, I have this following fmt() of a todo snippet:

```lua
fmt(
	        [=[
-- {}: {}
                ]=],
	{
		c(1, { t("TODO"), t("FIX"), t("HACK"), t("BUG") }),
		i(2, ""),
	})
```

My question is that if I'm at position 2 (Insert Node), can I somehow change the choice of the choice of the Choice Node at position 1?
Achieving this will make using choice nodes much more natural :smile: 

Another question I have is that is it possible to automatically jump to a specified position (index) after triggering the snippet? Maybe there's some option called "Initial Jump Position"?

Thank you very much!

**Comment by L3MON4D3** [comment]
Mhmmm, changing the choice from another place is theoretically possible, but not implemented :/
You'd have to re-write [this](https://github.com/L3MON4D3/LuaSnip/blob/3a814e069c73d49887b708417a9f58b948e807da/lua/luasnip/nodes/choiceNode.lua#L211-L289) function without the assumption that `current_node` is inside the choiceNode (make use of the restore_cursor-part, that will be necessary to move the selection/cursor if text was changed in front of it) (You'd also have to change the top-level call [here](https://github.com/L3MON4D3/LuaSnip/blob/f36fca2aa5df41d2f446a886de43e446e4648c64/lua/luasnip/init.lua#L370-L381)).

Jumping to a specific tabstop is possible with that same setting that allows immediately jumping to the $0, you should get away with replacing `insert_nodes[0]` with any other index.

**Reply by ziontee113** [reply]
Thank you for the answer. It's a more complicated matter than I thought 😊
